docker_cli/_docker_git.py

```diff
@@ -0,0 +1,77 @@
+import os
+from pathlib import Path
+from textwrap import dedent
+
+from ._docker_mamba import micromamba_docker_lines
+from ._url_reader import URLReader
+
+
+def git_extract_dockerfile(
+    base: str,
+    archive_url: str,
+    url_reader: URLReader,
+    folder_path: os.PathLike[str] = Path("repo"),
+) -> str:
+    """
+    Returns a Dockerfile-formatted string with instructions to fetch a Git archive.
+
+    Parameters
+    ----------
+    base : str
+        The base image for this dockerfile.
+    archive_url : str
+        The URL of the Git archive. Must be a `tar.gz` file.
+    url_reader : URLReader
+        The URL reader program to fetch the archive with.
+    folder_name : path-like, optional
```

Comment:
Let's rename "folder" to "directory", which I think is the more widely used term.

```suggestion
    directory : path-like, optional
```